frontend/src/components/viewers/Tensorboard.tsx

```diff
@@ -107,6 +113,9 @@ class TensorboardViewer extends Viewer<TensorboardViewerProps, TensorboardViewer
     // tensorboard instance on this pod.
     // TODO: figure out why the encoded pod address failed to open the tensorboard.
     const podAddress = this.state.podAddress.replace(/(^\w+:|^)\/\//, '');
+    if (podAddress && !this.state.tensorboardReady) {
+      this._checkTensorboardPodStatus(podAddress);
```

Comment:
It's not a good idea to put side effect into React `render`, because there's no guarantee when it runs. Every update on the top of the tree will trigger a render in this component so it can be too often, or there could be no change at all, so render is never triggered.

Using a `setInterval` in componentDidMount is the usual way to do so:
https://reactjs.org/docs/state-and-lifecycle.html#adding-lifecycle-methods-to-a-class

Reply:
Done.